.github/workflows/main.yml

```diff
@@ -0,0 +1,18 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v12
+      - uses: cachix/cachix-action@v8
+        with:
+          name: gi-gtk-declarative
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
```

Comment:
This won't work unless a repository secret is created in the Github repo settings, as per: https://nix.dev/tutorials/continuous-integration-github-actions.html